### PR TITLE
Determine KATZ score an sich & with bedtypes and diagnoses

### PR DESCRIPTION
**Pull request by zezke**
These changes are for the DependencyCategory and bug tickets for JDS.

### CODE REVIEW

**Review by alineVercruysse — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -19,46 +19,164 @@ const getCategory = (score, dementia) => {
...
+};
+
+const isIncontinent = score => {
+	const toilet = score.charAt(3);
+
+	return toilet >= 3;
+};
+
+const isIncontinentAndNeedHelpsWithTransferFoodOrToilet = score => {
+	// eslint-disable-next-line no-unused-vars
```
Comment (pietskrein) [inline index.js:59]: Why disable eslint instead of simply removing the unused vars?

File: index.d.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export type Category = 'O' | 'A' | 'B' | 'C' | 'D' | 'CD';
+export type BedType = 'ROB' | 'RVT' | 'CVK' | 'DVC' | 'NO';
```
Comment (pietskrein) [inline index.d.ts:2]: Personally I would prefer an enum for this

**Review by pietskrein — COMMENTED** [review]
2 small comments and a question: is there a flowchart for this? Seems a bit easier to review the functionality this way 😉

**Comment by zezke** [comment]
Diagrams:

katz-decision-tree-v2 werkelijke KATZ(1)
katz-decision-tree-v1

File: index.d.ts (hunk 2 above)
Comment (zezke) [reply index.d.ts:2]: I would agree, but this is a JS implementation, so not much use 🤷

File: index.js (hunk 1 above)
Comment (zezke) [reply index.js:59]: image

File: index.js [hunk 3]
```diff
@@ -19,46 +19,168 @@ const getCategory = (score, dementia) => {
...
+	return isIncontinentAndNeedsExtraHelp ? 'Cd' : 'D';
+};
+
+const getCategoryForBBed = (score, bedType, dementia, palliative) => {
+	if (bedType === 'DVC') {
+		if (palliative) {
+			return 'Fp';
+		}
+
+		return dementia ? 'D' : 'F';
```
Comment (pietskrein) [inline index.js:114]: The chart says - when not demented - this can be `F` or `Fd`, but I guess you know what you are doing

File: index.js [hunk 4]
```diff
@@ -19,46 +19,168 @@ const getCategory = (score, dementia) => {
...
+	return isIncontinent(score) ? 'Cd' : 'D';
+};
+
+const getCategoryForCdBed = (bedType, dementia, palliative) => {
+	if (bedType === 'DVC') {
+		if (palliative) {
+			return 'Fp';
+		}
+
+		return dementia ? 'D' : 'F';
```
Comment (pietskrein) [inline index.js:157]: Same comment as above

**Review by pietskrein — APPROVED** [review]
If you are sure of `F` instead of `Fd` this looks good to me.

**Pull request merged**